server/service/appconfig_test.go

```diff
@@ -399,6 +401,33 @@ func TestMissingMetadata(t *testing.T) {
 	assert.Contains(t, invalid.Error(), "either metadata or metadata_url must be defined")
 }
 
+func TestSSOValidationValidatesSchemaInMetadataURL(t *testing.T) {
+	var schemas []string
+	schemas = append(schemas, getURISchemas()...)
+	schemas = append(schemas, "asdfaklsdfjalksdfja")
+
+	for _, scheme := range schemas {
+		actual := &fleet.InvalidArgumentError{}
+		sut := fleet.AppConfig{
+			SSOSettings: &fleet.SSOSettings{
+				EnableSSO: true,
+				SSOProviderSettings: fleet.SSOProviderSettings{
+					EntityID:    "fleet",
+					IssuerURI:   "http://issuer.idp.com",
```

Comment:
Nit: IssuerURI is deprecated, it can be omitted (We'll remove it here: https://github.com/fleetdm/fleet/issues/10688)